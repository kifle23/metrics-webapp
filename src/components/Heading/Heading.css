.heading {
    position: sticky;
    top: 0;
    z-index: 2;
    background: url(../../assets/worldmap.png) no-repeat center center/cover;
    background-color: #fd5092;
    background-size: 100% 100%;
    height: 30vw;
    min-height: 150px;
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    text-shadow: 0 0 5px #36454F;
    box-shadow: 0 1px 5px rgba(85, 85, 85, 0.342);
}

.heading h1 {
    font-size: 2rem;
}

.filter {
    background-color: transparent;
    padding: 0.2rem 1rem;
    color: #36454F;
}

.filter option {
    color: #36454F;
}

.filter:hover {
    opacity: 1;
}