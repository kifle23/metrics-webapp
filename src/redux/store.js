import { configureStore } from '@reduxjs/toolkit';
import weatherReducer from './weatherSlice';

export default configureStore({
  reducer: {
    data: weatherReducer,
  },
});