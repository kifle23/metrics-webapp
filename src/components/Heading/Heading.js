import React from 'react';
import { useDispatch } from 'react-redux';
import PropTypes from 'prop-types';
import './Heading.css';
import { filterCountries } from '../../redux/weatherSlice';

const Heading = ({ title, number, home }) => {
  const dispatch = useDispatch();

  const regions = [
    { name: 'Africa', value: 'africa' },
    { name: 'Asia', value: 'asia' },
    { name: 'Europe', value: 'europe' },
    { name: 'North America', value: 'north-america' },
    { name: 'Oceania', value: 'oceania' },
    { name: 'South America', value: 'south-america' },
  ];

  const handleFilterChange = (event) => {
    dispatch(filterCountries(event.target.value));
  };

  return (
    <div className="heading">
      <h1>{title}</h1>

      {home ? (
        <select className="filter" defaultValue="all" onChange={handleFilterChange}>
          <option value="all" disabled>
            __Filter Countries by Region__
          </option>
          {regions.map((region) => (
            <option key={region.value} value={region.value}>
              {region.name}
            </option>
          ))}
        </select>
      ) : (
        <span>{number}</span>
      )}
    </div>
  );
};

Heading.propTypes = {
  title: PropTypes.string.isRequired,
  home: PropTypes.bool,
  number: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
};

Heading.defaultProps = {
  home: false,
  number: '',
};

export default Heading;
