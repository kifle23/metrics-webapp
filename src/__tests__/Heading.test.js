import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import store from '../redux/store';
import Heading from '../components/Heading/Heading';

describe('Heading component', () => {
  test('renders correctly', () => {
    const component = renderer.create(
      <Provider store={store}>
        <Router>
          <Heading title="weather" />
        </Router>
      </Provider>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
