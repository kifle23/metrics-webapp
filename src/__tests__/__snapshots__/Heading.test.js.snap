// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Heading component renders correctly 1`] = `
<div
  className="heading"
>
  <h1>
    weather
  </h1>
  <span />
</div>
`;

exports[`Heading component test Renders Heading 1`] = `
<div
  className="heading"
>
  <h1>
    weather
  </h1>
  <span />
</div>
`;
